app/services/fact_importer.rb

```diff
@@ -4,24 +4,6 @@ class FactImporter
   delegate :logger, :to => :Rails
   attr_reader :counters
 
-  def self.importer_for(type)
-    importers[type.to_s]
-  end
-
-  def self.importers
-    @importers ||= {}.with_indifferent_access
-  end
-
-  def self.register_fact_importer(key, klass, default = false)
-    importers.default = klass if default
-
-    importers[key.to_sym] = klass
-  end
-
-  def self.fact_features
-    importers.map { |_type, importer| importer.authorized_smart_proxy_features }.compact.flatten.uniq
-  end
-
```

Comment:
These should all get deprecation warnings and change implementation to use the regstry.